core/tx_pool.go

```diff
@@ -521,6 +522,7 @@ func (pool *TxPool) statsWithMinBaseFee(minBaseFee *big.Int) (int, int) {
 			pending++
 		}
 	}
+	realPendingGauge.Update(int64(pending))
```

Comment:
`StatsWithMinBaseFee` is only called from the GPO if the congestion threshold is enabled. 
So I think we have two options:
1. start thread that calls `StatsWithMinBaseFee` in whatever interval we'd like to have the data granularity + random, unpredictable calls from somewhere else
2. remove `realPendingGauge.Update(int64(pending))` here and only update it in separate thread

Reply:
makes sense to me too. to avoid duplication.